packages/rum/src/domain/segmentCollection/segmentCollection.spec.ts

```diff
@@ -214,6 +216,47 @@ describe('startSegmentCollection', () => {
       expect(sendSpy).toHaveBeenCalledTimes(1)
     })
   })
+
+  describe('updates view stats', () => {
+    it('when adding a new record', () => {
+      const { addRecord, worker } = startSegmentCollection(CONTEXT)
+      expect(getViewStats('b')).toEqual(undefined)
+      addRecord(RECORD)
+      worker.processAllMessages()
+      expect(getViewStats('b')).toEqual({
+        segments_count: 1,
+        records_count: 1,
+        segments_total_raw_size: 37,
+      })
+    })
+
+    it('when sending a segment', () => {
+      const { lifeCycle, addRecord, worker } = startSegmentCollection(CONTEXT)
+      expect(getViewStats('b')).toEqual(undefined)
+      addRecord(RECORD)
+      lifeCycle.notify(LifeCycleEventType.BEFORE_UNLOAD)
+      worker.processAllMessages()
+      expect(getViewStats('b')).toEqual({
+        segments_count: 1,
+        records_count: 1,
+        segments_total_raw_size: 193,
+      })
+    })
+
+    it('when creating a new segment', () => {
```

Comment:
I like to think of tests as being independant, but here this test depends on the one before it to give the expected results. is there a way to remove this implicit dependency ?

Reply:
Sorry, I was wrong, there is no dependency after all.
I didn't see the second call to `addRecord` here and thought it was counting the record from the previous test to give `records_count: 2`